src/braket/circuits/quantum_operator_helpers.py

```diff
@@ -23,7 +23,7 @@ def verify_quantum_operator_matrix_dimensions(matrix: np.array) -> None:
     raising `ValueError` otherwise.
 
     Args:
-        matrix (np.ndarray): matrix to verify
+        matrix (array): matrix to verify
```

Comment:
What is the array type ? np.ndarray tells me that it is a numpy array but this makes it ambiguous?

Reply:
Although it's often used, "np" is actually not a module.  I think we should document it as numpy.array or array. I chose the shorter option, since we often don't specify the modules for objects that we use, and I don't think that numpy should be an exception (what do you think?)

Reply:
I think np is pretty ubiquitous and also in scope, so I personally think np.ndarray is more clear for docs. Numpy just uses "ndarray" and "array_like" in their documentation, which could be another option too

Reply:
We actually use "array" here in the type hint, or I would have left the doc as "ndarray" as well. If it they really are ndarrays, we should change both.

Reply:
Oh good catch, ya I believe that should be ndarray